Fix rabbitmq_user when using force on user with permissions

sylr opened the pull request:
##### SUMMARY

On rabbitmq 3.7 using `force: yes` fails because outputs of rabbitmqctl gives an empty line

```
root@rabbitmq-vm1:~# rabbitmqctl -q list_user_permissions john

root@rabbitmq-vm1:~#
```

Provoking an error on

https://github.com/ansible/ansible/blob/456af458fcc6dfe29b89b8ab5a88ad1baea2fbe9/lib/ansible/modules/messaging/rabbitmq_user.py#L185

Because `perm.split('\t')` does not find any `\t`

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME
module rabbitmq_user

webknjaz commented on lib/ansible/modules/messaging/rabbitmq_user.py:
```diff
@@ -146,7 +146,7 @@ def _exec(self, args, run_in_check_mode=False):
             if self.node is not None:
                 cmd.extend(['-n', self.node])
             rc, out, err = self.module.run_command(cmd + args, check_rc=True)
-            return out.splitlines()
```
I think it's more like:
```python
return out.strip().splitlines()
```
which will have same result

sylr replied:
indeed

Commit:
Fix rabbitmq_user when using force on user with permissions

On rabbitmq 3.7 using `force: yes` fails because outputs of rabbitmqctl gives an empty line

```
root@rabbitmq-vm1:~# rabbitmqctl -q list_user_permissions john

root@rabbitmq-vm1:~#
```

Provoking an error on

https://github.com/ansible/ansible/blob/456af458fcc6dfe29b89b8ab5a88ad1baea2fbe9/lib/ansible/modules/messaging/rabbitmq_user.py#L185

Because `perm.split('\t')` does not find any `\t`

Signed-off-by: Sylvain Rabot <s.rabot@lectra.com>

sylr commented:
@webknjaz can we merge this ?

sylr commented:
Anyone ? @dagwieers ?

sylr commented:
@dagwieers @gundalow @webknjaz guys ?

hyperized commented:
@sylr help me understand please. Do you have a playbook where this bug shows? What is the strip supposed to do exactly?

dagwieers commented:
I am not involved in RabbitMQ. If you like to speed up RabbitMQ decision-making it could help to find other RabbitMQ contributors for discussing and merging RabbitMQ-related PRs. This works well for other technologies.

https://github.com/ansible/community/wiki/RabbitMQ

webknjaz commented:
@sylr this is a change to the community-supported code. As Dag has outlined earlier, the decision is up to the maintainers and/or working group who is more knowledgeable in this area.
Here you can check who is on the related maintainers team: https://github.com/ansible/ansible/blob/devel/.github/BOTMETA.yml#L262

sylr commented:
@hyperized The strip remove the empty lines output by rabbitmqctl when you list permissions of a user which does not have any. This empty lines messes up the logic because it try to iterate on an empty line.

hyperized commented:
@sylr any chance you can add a test to this code to confirm the change is effective?

sylr commented:
I'm afraid I won't have time to do that sorry. But it works, we've been using this patch for a while now at our company.

sylr committed:
Merge branch 'devel' into rabbitmq
